### ISSUE TITLE
Charting support

### ISSUE THREAD

**Issue by igitur** [ClosedXML/ClosedXML#50]

**Comment by mihaj** [comment]
This feature is not yet available, right? Is there a plan to implement this?

**Comment by vbjay** [comment]
You do know how open source  projects work, right?  Some people suggest
features. Some people create the features. If you want the feature so bad
then help make it.

**Comment by igitur** [comment]
At the moment, I'm focused on squashing bugs and getting the library stable and efficient. Adding charting support will be a big change and needs planning. That said, as @vbjay mentions, we'd appreciate your help and any pull requests will be very helpful.

**Comment by mihaj** [comment]
OK guys. I understand. I will check the library, any specific thing to test?

**Comment by LievenDeFoor** [comment]
Is there any way to use OpenXML to create a chart on a ClosedXML workbook?
Or to expose the underlying OpenXML workbook (which I assume is living in the ClosedXML workbook) and use OpenXML on that?

**Comment by igitur** [comment]
> Is there any way to use OpenXML to create a chart on a ClosedXML workbook?
> Or to expose the underlying OpenXML workbook (which I assume is living in the ClosedXML workbook) and use OpenXML on that?

Not while ClosedXML has the workbook open. You can save the workbook in ClosedXML, release the `XLWorkbook` instance and then use OpenXML to open and modify it to add a charge, if you wish.

**Comment by LievenDeFoor** [comment]
> 
> 
> > Is there any way to use OpenXML to create a chart on a ClosedXML workbook?
> > Or to expose the underlying OpenXML workbook (which I assume is living in the ClosedXML workbook) and use OpenXML on that?
> 
> Not while ClosedXML has the workbook open. You can save the workbook in ClosedXML, release the `XLWorkbook` instance and then use OpenXML to open and modify it to add a charge, if you wish.

So if I'm understanding correctly there is no OpenXML workbook behind the scenes while building the Excel document using ClosedXML? Only when Save() is called, OpenXML API is called and the ClosedXML internal structure in used to create an OpenXML document?
Or is there an inner OpenXML workbook that could somehow get exposed during the lifetime of the ClosedXML workbook and be manipulated directly (operating on the same instance) through OpenXML?

**Comment by igitur** [comment]
That's correct. When loading an existing file, ClosedXML reads all the contents (via OpenXML) into memory. The OpenXML "package" is then closed. Editing in ClosedXML is done in-memory and when saving OpenXML is used again. Other packages, like EPPlus work directly and immediately on the underlying OpenXML package. IMHO, this is the core reason why the ClosedXML API is more fluent/untuitive but lacks feature support compared to other libraries.
